.github/workflows/license-check.yml

```diff
@@ -0,0 +1,45 @@
+---
+name: License Check
+
+"on":
+  pull_request:
+    paths:
+      - '**/pyproject.toml'
+      - '.github/workflows/license-check.yml'
+      - '.github/workflows/scripts/check_licenses.py'
+
+jobs:
+  check-licenses:
+    name: Check Package Licenses
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.10'
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install tomli requests urllib3
+
+      - name: Check licenses
+        run: |
+          python .github/workflows/scripts/check_licenses.py \
+            pyproject.toml || exit_code=$?
+          if [ "${exit_code:-0}" -eq 1 ]; then
```

Comment:
Shall we use the condition of "the exit code is not 0"  just in case the script returns some error with exit code other than 1?  also set the default value to 1 if exit_code does not exist

eg. `if [ "${exit_code:-1}" -ne 0 ]`